service/src/main/java/org/apache/griffin/core/metastore/hive/HiveMetaStoreServiceJdbcImpl.java

```diff
@@ -0,0 +1,287 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Qualifier(value = "hive_jdbc")
+@CacheConfig(cacheNames = "jdbchive", keyGenerator = "cacheKeyGenerator")
+public class HiveMetaStoreServiceJdbcImpl implements HiveMetaStoreService {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(HiveMetaStoreService.class);
+
+    @Value("${hive.jdbc.className}")
+    private String hiveClassName;
+
+    @Value("${hive.jdbc.url}")
+    private String hiveUrl;
+
+    private Connection conn;
+
+    public void setConn(Connection conn) {
+        this.conn = conn;
+    }
+
+    public void setHiveClassName(String hiveClassName) {
+        this.hiveClassName = hiveClassName;
+    }
+
+    @Override
+    @Cacheable(unless = "#result==null")
+    public Iterable<String> getAllDatabases() {
+        String sql = "show databases";
+        return queryHiveString(sql);
+    }
+
+    @Override
+    @Cacheable(unless = "#result==null")
+    public Iterable<String> getAllTableNames(String dbName) {
+        String sql = "show tables in " + dbName;
+        return queryHiveString(sql);
+    }
+
+    @Override
+    @Cacheable(unless = "#result==null")
+    public Map<String, List<String>> getAllTableNames() {
+        // Not recommend this method because this method will generate one query for every database
+        // If there has a lots of databases in Hive, this method will lead to Griffin crash
+        Map<String, List<String>> res = new HashMap<>();
+
+        for (String dbName : getAllDatabases()) {
+            List<String> list = (List<String>) queryHiveString("show tables in " + dbName);
+            res.put(dbName, list);
+        }
+
+        return res;
+    }
+
+    @Override
+    public List<Table> getAllTable(String db) {
+        return null;
+    }
+
+    @Override
+    public Map<String, List<Table>> getAllTable() {
+        return null;
+    }
+
+    @Override
+    @Cacheable(unless = "#result==null")
+    public Table getTable(String dbName, String tableName) {
+        Table result = new Table();
+        result.setDbName(dbName);
+        result.setTableName(tableName);
+
+        try {
+            Class.forName(hiveClassName);
+            if (conn == null) {
+                conn = DriverManager.getConnection(hiveUrl);
+            }
+        } catch (ClassNotFoundException | SQLException e) {
+            e.printStackTrace();
+        }
+
```

Comment:
1.Pls use `LOGGER.error ` instead of `e.printStackTrace()`
2.If exception occurs, here just catches it and doesn't do anything. Also, the next query is still to report exceptions. 
May be register class name and hive connection be done after class created, like 
```@PostConstruct
    public void init() {
        Class.forName(hiveClassName);
        if (conn == null) {
            conn = DriverManager.getConnection(hiveUrl);
        }
    }
```
you can also add retry to connect hive.
This is just my advice.

Reply:
It's not easy to use Mockito to do the unit test in this way. I will try to use Hive connection pool to replace this solution in future.